addons/account/models/account_move.py

```diff
@@ -1462,7 +1462,7 @@ def _inverse_amount_total(self):
             to_write = []
 
             amount_currency = abs(move.amount_total)
-            balance = move.currency_id._convert(amount_currency, move.company_currency_id, move.company_id, move.date)
+            balance = move.currency_id._convert(amount_currency, move.company_currency_id, move.company_id, move.invoice_date or move.date)
```

Comment:
Instead of:
```
balance = move.currency_id._convert(amount_currency, move.company_currency_id, move.company_id, move.invoice_date or move.date)
```

This should have been:
```
balance = move.currency_id._convert(amount_currency, move.company_currency_id, move.company_id, move.date or move.invoice_date)
```

And so on.

Regards.

Reply:
@hbto Not so according to @tsb-odoo (maybe you can discuss with him about your l10n_mx_edi_decimal_places as well)  So, thanks for pointing this out as it seems ok.  @aliyatastemirova could you check as well and see if we need a PR?

Reply:
I'm checking with tsb